### PR TITLE
Fix type hints and replace deprecated method usage

### PR DESCRIPTION
**Pull request by avstudnitz**
### Description (*)
Updates type hints of return value and replaces deprecated method calls like `addSuccess` with the newer methods like `addSuccessMessage`.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @avstudnitz. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-4409](https://jira.corp.magento.com/browse/ENGCOM-4409) has been created to process this Pull Request

**Comment by soleksii** [comment]
:heavy_check_mark: QA Passed

File: app/code/Magento/Newsletter/Controller/Subscriber/Confirm.php [hunk 1]
```diff
@@ -4,12 +4,19 @@
...
  * See COPYING.txt for license details.
  */
+
 namespace Magento\Newsletter\Controller\Subscriber;
 
-class Confirm extends \Magento\Newsletter\Controller\Subscriber
+use Magento\Framework\App\Action\HttpGetActionInterface;
+
+/**
+ * Confirm subscritpion controller.
```
Comment (orlangur) [inline app/code/Magento/Newsletter/Controller/Subscriber/Confirm.php]: @nmalevanec please fix typo.

Comment (nmalevanec) [reply app/code/Magento/Newsletter/Controller/Subscriber/Confirm.php]: Done. Thanks for noticing.

**Pull request merged**